### DISCUSSION TITLE
"ext_id" on the VOD Information always a mismatch.

### DISCUSSION THREAD

**Discussion by CappiSteijns** [Zibbp/ganymede#345] [General]
Wouldn't count this as an issue because beside archiving and watching on Ganymede I'm using the VOD Information from ganymede to make an index on my website so other people can easily find vods trough playlists I made in Ganymede.

For example: https://lekkerspelenfandom.nl/untitled-2/

But take this VOD information as example:

```
{
  "id": "85f6251d-99f0-11ee-9e8f-0242ac140003",
  "ext_id": "42114447897",
  "platform": "twitch",
  "type": "live",
  "title": "PETER VS TIMON 2023 🏆",
  "duration": 9534,
  "views": 1,
  "resolution": "best",
  "thumbnail_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-thumbnail.jpg",
  "web_thumbnail_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-web_thumbnail.jpg",
  "video_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-video.mp4",
  "chat_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-chat.json",
  "chat_video_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-chat.mp4",
  "info_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-info.json",
  "folder_name": "2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003",
  "file_name": "42114447897",
  "local_views": 1,
  "streamed_at": "2023-12-13T19:48:01.377099Z",
  "updated_at": "2024-01-07T03:13:33.711539Z",
  "created_at": "2023-12-13T19:48:01.377256Z",
  "edges": {
    "channel": {
      "id": "da865789-421d-445f-8fc2-817fd2446111",
      "ext_id": "52385053",
      "name": "lekkerspelen",
      "display_name": "lekkerspelen",
      "image_path": "/vods/lekkerspelen/profile.png",
      "updated_at": "2023-02-01T10:50:34.954278Z",
      "created_at": "2023-01-22T15:19:14.945367Z",
      "edges": {}
    }
  }
}
```
You would think the "ext_id" would be the id of the vod? So that would be https://www.twitch.tv/videos/42114447897.
But that doesn't exist, it actually is in this case: https://www.twitch.tv/videos/2003234717.

So I was thinking that maybe twitch changes the "ext_id" when it's going from live to vod?

Probably out of scope for this project but was wondering if there was an fix for this.

**Comment by Zibbp** [comment]
Twitch has a VOD ID and a live ID which is why they are different. Once the video download is complete, I can attempt to update the Ganymede VOD with the Twitch VOD ID that corresponds with the live stream. This obviously has some catches. The VOD needs to be available immediately so the channel needs to not have the manual vod creation setting enabled (where the channel needs to click a button to publish the vod), and needs vods turned on in general.

If the Ganymede livestream video does get updated with the correct external ID of the VOD then you won't be able to archive the VOD as it technically already exists in Ganymede (the external IDs are the same). Probably not a huge concern as you already have the livestream. 

Would this solve your issue?

**Reply by Zibbp** [reply]
Made an issue to track this better https://github.com/Zibbp/ganymede/issues/346
